pulpcore/cli/common/generic.py

```diff
@@ -296,6 +296,12 @@ def pulp_group(
 
 
 class PulpOption(click.Option):
+    """
+    Wrapper class for `click.Option`.
+
+    This provides a standard way of defining CLI options and helps keep consistent behavior.
```

Comment:
We want to encourage using the factory, so please mention that here.
Also this is a great place to tell, what `PulpOption` provides on top of `click.Option`.